x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/package-info.java

```diff
@@ -0,0 +1,154 @@
+ *         {@code<pre>
+ * [remote "elastic"]
+ * url = git@github.com:elastic/elasticsearch.git
+ * fetch = +refs/heads/*:refs/remotes/elastic/*
+ * [remote "nik9000"]
+ * url = git@github.com:nik9000/elasticsearch.git
+ * fetch = +refs/heads/*:refs/remotes/nik9000/*
+ *         </pre>}
+ *     </li>
+ *     <li>
+ *         Feel free to use {@code git} as a scratch pad. We're going to squash all commits
+ *         before merging and will only keep the PR subject line and description in the
+ *         commit message.
+ *     </li>
+ *     <li>
+ *         Open Elasticsearch in IntelliJ.
+ *     </li>
+ *     <li>
+ *         Open {@code x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/CsvTests.java}
+ *         and run it. IntelliJ will take a few minutes to compile everything but the test itself
+ *         should take only a few seconds. This is a fast path to running ESQL's integration tests.
+ *     </li>
+ *     <li>
+ *         Pick one of the csv-spec files in {@code x-pack/plugin/esql/qa/testFixtures/src/main/resources/}
+ *         and add a test for the function you want to write. These files are roughly themed but there
+ *         isn't a strong guiding principle in the theme.
+ *     </li>
+ *     <li>
+ *         Rerun the {@code CsvTests} and watch your new test fail. Yay, TDD doing it's job.
+ *     </li>
+ *     <li>
+ *         Find a function in this package similar to the one you are working on and copy it to build
+ *         yours. There's come ceremony required in each function class to make it constant foldable
+ *         and return the right types. Take a stab at these, but don't worry too much about getting
+ *         it right.
+ *     </li>
+ *     <li>
+ *         There are also methods annotated with {@link org.elasticsearch.compute.ann.Evaluator}
+ *         that contain the actual inner implementation of the function. Modify those to look right
+ *         and click {@code Build->Recompile 'FunctionName.java'} in IntelliJ. This should generate
+ *         an {@link org.elasticsearch.compute.operator.EvalOperator.ExpressionEvaluator} implementation
+ *         calling the method annotated with {@link org.elasticsearch.compute.ann.Evaluator}.
+ *     <li>
+ *         Once your evaluator is generated you can implement {@link org.elasticsearch.xpack.esql.planner.Mappable#toEvaluator},
+ *         having it return the generated evaluator.
+ *     </li>
+ *     <li>
+ *         Add your function to {@link org.elasticsearch.xpack.esql.expression.function.EsqlFunctionRegistry}.
+ *         This links it into the language and {@code SHOW FUNCTIONS}. Also add your function to
+ *         {@link org.elasticsearch.xpack.esql.io.stream.PlanNamedTypes}. This makes your function
+ *         serializable over the wire. Mostly you can copy existing implementations for both.
+ *     </li>
+ *     <li>
+ *         Rerun the {@code CsvTests}. They should find your function and maybe even pass. Add a
```

Comment:
Imo, we should instruct authors to also consider one-two more complex tests with the newly added functions (functions passed as parameters to other functions for example), these types of queries can sometimes reveal some flaws in optimization rules, or indicate that some core methods of that specific function were not implemented/overriden or poorly implemented.